game-core/src/test/java/games/strategy/triplea/ai/BattleCalculatorComparisonTests.java

```diff
@@ -0,0 +1,395 @@
+  private static final UnitType CRUISER =
+      checkNotNull(data.getUnitTypeList().getUnitType("cruiser"));
+  private static final UnitType BATTLESHIP =
+      checkNotNull(data.getUnitTypeList().getUnitType("battleship"));
+
+  private static final GameData dataTww = TestMapGameData.TWW.getGameData();
+  private static final GamePlayer BRITAIN = checkNotNull(GameDataTestUtil.britain(dataTww));
+  private static final GamePlayer GERMANY = checkNotNull(GameDataTestUtil.germany(dataTww));
+  private static final Territory LAND_NO_ATTACHMENTS =
+      checkNotNull(GameDataTestUtil.territory("Alberta", dataTww));
+
+  @Test
+  void comparePerformanceOfCalculators() throws InterruptedException {
+    // System.out.println("Sleeping to allow attaching asyncProfiler");
+    // Thread.sleep(2000);
+    if (false) {
+      final int quantity = 3;
+      System.out.println(quantity + " of each Total World War land units");
+      final Collection<Unit> attackingUnits = new ArrayList<>();
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishInfantry").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishAlpineInfantry").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishMarine").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishArtillery").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishHeavyArtillery").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww
+              .getUnitTypeList()
+              .getUnitType("britishMobileArtillery")
+              .create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishMech.Infantry").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishTank").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishHeavyTank").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishFighter").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww
+              .getUnitTypeList()
+              .getUnitType("britishAdvancedFighter")
+              .create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishTacticalBomber").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww
+              .getUnitTypeList()
+              .getUnitType("britishAdvancedTacticalBomber")
+              .create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishAntiAirGun").create(quantity, BRITAIN));
```

Comment:
Perhaps you can make a helper for this logic that just gets passed the unit type, country and quantity? To avoid the repetitiveness and extra wrapping.